src-tauri/src/systemtray_manager.rs

```diff
@@ -0,0 +1,239 @@
+                    format_hashrate(data.cpu_hashrate),
+                    format_hashrate(data.gpu_hashrate),
+                    format_currency(data.estimated_earning / 1_000_000.0, "tXTM/day")
+                )
+            }
+        }
+    }
+
+    pub fn initialize_tray(&mut self, app: AppHandle) -> Result<(), anyhow::Error> {
+        let tray = app.tray_by_id("universe-tray-id").expect("tray not found");
+        let menu = self.initialize_menu(app.clone())?;
+        tray.set_menu(Some(menu.clone()))?;
+
+        tray.on_menu_event(move |app, event| match event.id.as_ref() {
+            "minimize_toggle" => {
+                let window = match app.get_webview_window("main") {
+                    Some(window) => window,
+                    None => {
+                        error!(target: LOG_TARGET, "Failed to get main window");
+                        return;
+                    }
+                };
+
+                if window.is_minimized().unwrap_or(false) {
+                    info!(target: LOG_TARGET, "Unminimizing window");
+                    match PlatformUtils::detect_current_os() {
+                        CurrentOperatingSystem::Linux => {
+                            window.hide().unwrap_or_else(|error| error!(target: LOG_TARGET, "Failed hide window: {}", error));
+                            window.unminimize().unwrap_or_else(|error| error!(target: LOG_TARGET, "Failed to unminimize window: {}", error));
+                            window.show().unwrap_or_else(|error| error!(target: LOG_TARGET, "Failed to show window: {}", error));
+                            window.set_focus().unwrap_or_else(|error| error!(target: LOG_TARGET, "Failed to set focus on window: {}", error));
+                        }
+                        _ => {
+                            window.unminimize().unwrap_or_else(|error| {
+                                error!(target: LOG_TARGET, "Failed to unminimize window: {}", error);
+                            });
+                            window.set_focus().unwrap_or_else(|error| {
+                                error!(target: LOG_TARGET, "Failed to set focus on window: {}", error);
+                            });
+                        }
+                    }
+                } else {
+                    info!(target: LOG_TARGET, "Minimizing window");
+                    window.minimize().unwrap_or_else(|error| {
+                        error!(target: LOG_TARGET, "Failed to minimize window: {}", error);
+                    });
+                }
+            },
+            _ => {
+                println!("menu item {:?} not handled", event.id);
+            }
+        });
+
+        self.menu.replace(menu);
+        self.tray.replace(tray);
+
+        Ok(())
+    }
+
+    pub fn update_tray(&mut self, data: SystemTrayData) {
+        if let Some(tray) = &self.tray {
+            if let Err(e) = tray.set_tooltip(Some(self.get_tooltip_text(data.clone()))) {
```

Comment:
Maybe `get_tooltip_text` should return an `Option<String>` ? 

Only if it makes the code simpler. It means `get_tooltip_text` could return None for linux. And if set_tooltip takes an option then everything should flow nicely.